opennms-dao/src/main/java/org/opennms/netmgt/dao/support/AbstractTopLevelResourceType.java

```diff
@@ -47,7 +48,10 @@ public boolean isResourceTypeOnParent(OnmsResource parent) {
 
     @Override
     public List<OnmsResource> getResourcesForParent(OnmsResource parent) {
-        return getTopLevelResources();
+        if (parent == null) {
```

Comment:
I understand, but the parameter is never used. Having this in mind, is it correct to assume that there are implementations of AbstractTopLevelResourceType that override this method and use the provided parameter ?